File: python/problem7.py
import math

def isPrime(num):
    if num == 1: return False
    for x in range(2,int(math.sqrt(num))+1):
        if num % x == 0: return False

    return True

def generateNthPrimeNumber(limit):
    count = 1
    n = 3
    while True:
        if isPrime(n): count += 1
        if count == limit: break
        n += 1
    return n

if __name__=='__main__':
    print(generateNthPrimeNumber(6))

File: go/problem7.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(GenerateNthPrimeNumber(10))
}

func GenerateNthPrimeNumber(limit int) int {

	count := 1
	n := 3
	for {
		if isPrime(n) {
			count++
			if count == limit {
				break
			}
		}
		n += 2
	}
	return n
}

func isPrime(num int) bool {

	for i := 2; i <= int(math.Sqrt(float64(num))); i++ {
		if num%i == 0 {
			return false
		}
	}

	return true
}
